yt_dlp/extractor/youtube.py

```diff
@@ -3634,6 +3634,8 @@ def _extract_comment(self, entities, parent=None):
                 'author_thumbnail': ('author', 'avatarThumbnailUrl', {url_or_none}),
                 'author_is_uploader': ('author', 'isCreator', {bool}),
                 'author_is_verified': ('author', 'isVerified', {bool}),
+                'author_is_member': ('author', 'sponsorBadgeA11y', {str}),
+                'author_member_badge': ('author', 'sponsorBadgeUrl', {url_or_none}),
```

Comment:
What are the possible values for `author_is_member`? A string of "true" or "false"? Conventionally, if the field name is in the format of`x_is_y`, its value should be boolean

How many different badges are there? Is it like a silver/gold/platinum type thing?

Also, if we add these fields for the Youtube extractor, then we need to document them in the InfoExtractor docstring:

https://github.com/yt-dlp/yt-dlp/blob/330335386d4f7603d92d6796798375336005275e/yt_dlp/extractor/common.py#L339-L349

I think `author_is_member` is general enough that other sites could make use of it, but I don't know how generally useful `author_member_badge` would be

Reply:
The possible values for author is member are `Member (XX months XX years)` (or `Member (XX months)` for users under a year and `Member (XX years)` for member's who's subscribed for that many years exactly). The field is a string and not a boolean. Either they have a current membership and it's displayed, or they do not. Hence, if the string is there, they are a member, if it's not, they are not. The field is not a boolean, and if it were, it simply omits how long the author has been a member. I feel splitting this into two fields "Is a member" and "member for how long" is redundant.

The badges are also channel dependent, where they are a URL to a custom image uploaded by the channel which correspond to the number of months as a member. `New Member`, `Month 1`, `Month 2`, `Month 6`, `Month 12`, `Month 12`, `Month 36`, and `Month 48` are the highest values I see in the memberships I have. If there is no string, then it is possible the channel does not have memberships available.

> I don't know how generally useful author_member_badge would be

This is already being archived in Live Chat data as `liveChatAuthorBadgeRenderer` as `customThumbnail` . The `author_is_member` is equivalent to the `label` and `tooltip` of that badge.

Reply:
Membership has ended, and the badges have turned into what appear to be default/generic ones. So as long as the user had memberships going, I think the badge urls as an identifier is worth archiving.